.github/workflows/codeql.cpp.yml

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: CodeQL - C++
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+    paths:
+      - .github/workflows/codeql.cpp.yml
+      - '**.c'
+      - '**.cc'
+      - '**.cpp'
+      - '**.h'
```

Comment:
Two other fairly common filename suffixes for C++ source files:

- `*.cxx`
- `*.hpp`

although I think I've only seen the latter on Windows platforms.

Reply:
> Two other fairly common filename suffixes for C++ source files:
> 
>     * `*.cxx`
> 
>     * `*.hpp`

Added in 1d1eba4d1c

> although I think I've only seen the latter on Windows platforms.

The ATS repo has one